solutions/swb-reference/src/services/dataSetService.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CreateProvisionDatasetRequest } from '@aws/swb-app/lib/datasets/createProvisionDatasetRequest';
+import { DataSet } from '@aws/swb-app/lib/datasets/dataSet';
+import { DatasetPluginInterface } from '@aws/swb-app/lib/datasets/datasetPluginInterface';
+import { DataSetsStoragePluginInterface } from '@aws/swb-app/lib/datasets/dataSetsStoragePluginInterface';
+import { AuditService } from '@aws/workbench-core-audit';
+import {
+  DataSetMetadataPlugin,
+  DataSetService as WorkbenchDataSetService
+} from '@aws/workbench-core-datasets';
+import { LoggingService } from '@aws/workbench-core-logging';
+
+export class DataSetService implements DatasetPluginInterface {
+  public readonly storagePlugin: DataSetsStoragePluginInterface;
+  private _workbenchDataSetService: WorkbenchDataSetService;
+
+  public constructor(
+    storagePlugin: DataSetsStoragePluginInterface,
+    audit: AuditService,
+    log: LoggingService,
+    masterDbProvider: DataSetMetadataPlugin
+  ) {
+    this._workbenchDataSetService = new WorkbenchDataSetService(audit, log, masterDbProvider);
+    this.storagePlugin = storagePlugin;
+  }
+
+  public addDataSetExternalEndpoint(
+    dataSetId: string,
```

Comment:
We probably want to create an interface for these params as well, just like how we have `CreateProvisionDatasetRequest`. I'm not sure why the DS package didn't have an interface for this request already.